c/tskit/trees.c

```diff
@@ -2734,6 +2734,56 @@ tsk_treeseq_divergence(tsk_treeseq_t *self, tsk_size_t num_sample_sets,
     return ret;
 }
 
+static int
+relatedness_summary_func(size_t state_dim, double *state, size_t result_dim,
+    double *result, void *params)
+{
+    sample_count_stat_params_t args = *(sample_count_stat_params_t *) params;
+    const double *x = state;
+    tsk_id_t i, j;
+    size_t k;
+    double c = 0;
+    double sumx = 0;
+    double num = 0;
+
+    for (k = 0; k < state_dim; k++) {
+        sumx += x[k];
+    }
+
+    for (k = 0; k < result_dim; k++) {
+        num += args.sample_set_sizes[k];
+    } 
+
+    if (num != sumx) {
+        c = 1;
+    }
```

Comment:
maybe the problem is floating-point error? A maybe easier way to do this (that might help?) would be to leave off `sums` and `num` and do
```
     for (k = 0; k < state_dim; k++) {
        if (x[k] != args.sample_set_sizes[k]) {
            c = 1;
        }
    }
```